os-server/src/main/java/app/openschool/user/UserServiceImpl.java

```diff
@@ -206,7 +206,15 @@ public Course deleteCourse(Long userId, Long courseId) {
   @Transactional
   public Course enrollCourse(User user, long courseId) {
     Course course = courseRepository.findById(courseId).orElseThrow(IllegalArgumentException::new);
-    user.getEnrolledCourses().add(CourseMapper.toEnrolledCourse(course, user));
+    Set<EnrolledCourse> enrolledCourses = user.getEnrolledCourses();
+    boolean enrolledCourseExists =
+        enrolledCourses.stream()
+            .map(enrolledCourse -> enrolledCourse.getCourse().getId())
+            .anyMatch(fetchedCourseId -> fetchedCourseId == courseId);
+    if (enrolledCourseExists) {
+      throw new IllegalArgumentException();
```

Comment:
lets throw some meaningful exception, so the user on UI can see what's the problem